File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        unordered_map<int,int>mp;
        int n = nums.size();
        for(int i=n-1;i>=0;i--){
            if(mp.count(nums[i])!=0){
                return i/3 + 1;
            }
            mp[nums[i]]++;
        }
        return 0;
    }
};

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumOperations = function(nums) {
    var mp =[];
    for(let i = nums.length-1;i>=0;i--){
        if(mp[nums[i]]){
            var temp = i/3;
            return parseInt(temp) + 1;
        }
        mp[nums[i]]=1;
    }
    return 0;
};